pkg/process/procutil/process_linux_test.go

```diff
@@ -0,0 +1,186 @@
+// +build linux
+
+package procutil
+
+import (
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DataDog/gopsutil/process"
+)
+
+func TestGetActivePIDs(t *testing.T) {
+	os.Setenv("HOST_PROC", "resources/test_procfs/proc")
+	defer os.Unsetenv("HOST_PROC")
+
+	probe := NewProcessProbe()
+	defer probe.Close()
+
+	actual, err := probe.getActivePIDs()
+	assert.NoError(t, err)
+
+	expect, err := process.Pids()
+	assert.NoError(t, err)
+
+	assert.ElementsMatch(t, expect, actual)
+}
+
+func TestTrimAndSplitBytes(t *testing.T) {
+	for _, tc := range []struct {
+		input  []byte
+		expect []string
+	}{
+		{
+			input:  []byte{115, 115, 104, 100, 58, 32, 115, 117, 110, 110, 121, 46, 107, 108, 97, 105, 114, 64, 112, 116, 115, 47, 48},
+			expect: []string{string([]byte{115, 115, 104, 100, 58, 32, 115, 117, 110, 110, 121, 46, 107, 108, 97, 105, 114, 64, 112, 116, 115, 47, 48})},
+		},
+		{
+			input:  []byte{40, 115, 100, 45, 112, 97, 109, 41, 0, 0, 0},
+			expect: []string{string([]byte{40, 115, 100, 45, 112, 97, 109, 41})},
+		},
+		{
+			input:  []byte{115, 115, 104, 100, 58, 32, 118, 97, 103, 114, 97, 110, 116, 64, 112, 116, 115, 47, 48, 0, 0},
+			expect: []string{string([]byte{115, 115, 104, 100, 58, 32, 118, 97, 103, 114, 97, 110, 116, 64, 112, 116, 115, 47, 48})},
+		},
+		{
+			input: []byte{47, 117, 115, 114, 47, 98, 105, 110, 47, 100, 111, 99, 107, 101, 114, 100, 0, 45, 72, 0, 102, 100, 58, 47, 47, 0},
+			expect: []string{
+				string([]byte{47, 117, 115, 114, 47, 98, 105, 110, 47, 100, 111, 99, 107, 101, 114, 100}),
+				string([]byte{45, 72}),
+				string([]byte{102, 100, 58, 47, 47}),
+			},
+		},
+		{
+			input: []byte{47, 117, 115, 114, 47, 98, 105, 110, 47, 100, 111, 99, 107, 101, 114, 100, 0, 45, 72, 0, 102, 100, 58, 47, 47},
+			expect: []string{
+				string([]byte{47, 117, 115, 114, 47, 98, 105, 110, 47, 100, 111, 99, 107, 101, 114, 100}),
+				string([]byte{45, 72}),
+				string([]byte{102, 100, 58, 47, 47}),
+			},
+		},
+		{
+			input:  []byte{0, 0, 47, 115, 98, 105, 110, 47, 105, 110, 105, 116, 0},
+			expect: []string{string([]byte{47, 115, 98, 105, 110, 47, 105, 110, 105, 116})},
+		},
+	} {
```

Comment:
Should we add a test case for consecutive null bytes in the middle of the command line?

Reply:
Yes, I added a test case above with multiple 0s.